spec.emu

```diff
@@ -732,12 +890,62 @@ contributors: Luca Casonato, Guy Bedford, Nicolò Ribaudo
+          HostLoadImportedModule (
+            _referrer_: a Script Record, a Cyclic Module Record, or a Realm Record,
+            <del>_specifier_: a String,</del>
+            <ins>_moduleRequest_: a ModuleRequest Record,</ins>
+            _hostDefined_: anything,
+            _payload_: a GraphLoadingState Record or a PromiseCapability Record,
+          ): ~unused~
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd></dd>
+        </dl>
+
+        <emu-note id="note-HostLoadImportedModule-referrer-Realm-Record">
+          <p>An example of when _referrer_ can be a Realm Record is in a web browser host. There, if a user clicks on a control given by</p>
+
+          <pre><code class="html">&lt;button type="button" onclick="import('./foo.mjs')"&gt;Click me&lt;/button&gt;</code></pre>
+
+          <p>there will be no active script or module at the time the <emu-xref href="#sec-import-calls">`import()`</emu-xref> expression runs. More generally, this can happen in any situation where the host pushes execution contexts with *null* ScriptOrModule components onto the execution context stack.</p>
+        </emu-note>
+
+        <p>An implementation of HostLoadImportedModule must conform to the following requirements:</p>
+        <ul>
+          <li>
+            The host environment must perform FinishLoadingImportedModule(_referrer_, <del>_specifier_</del><ins>_moduleRequest_</ins>, _payload_, _result_), where _result_ is either a normal completion containing the loaded Module Record or a throw completion, either synchronously or asynchronously.
+          </li>
+          <li>
+            <p>If this operation is called multiple times with <del>the same</del><ins>two</ins> (_referrer_, <del>_specifier_</del><ins>_moduleRequest_</ins>) pair<ins>s</ins> such that:</p>
+            <ul>
+              <li><ins>the first _referrer_ is the same as the second _referrer_;</ins></li>
+              <li><ins>the first _moduleRequest_.[[Specifier]] is the same as the second _moduleRequest_.[[Specifier]];</ins></li>
+              <li>it performs FinishLoadingImportedModule(_referrer_, <del>_specifier_</del><ins>_moduleRequest_</ins>, _payload_, _result_) where _result_ is a normal completion,</li>
+            </ul>
+            <p>then it must perform FinishLoadingImportedModule(_referrer_, <del>_specifier_</del><ins>_moduleRequest_</ins>, _payload_, _result_) with the same _result_ each time.</p>
+          </li>
+          <li>
+            The operation must treat _payload_ as an opaque value to be passed through to FinishLoadingImportedModule.
+          </li>
+          <li>
+            <p><ins>When `[[Phase]]` is set to ~source~, this should not affect the identity or state of the resultant Module Record as if it had been called for the ~evaluation~ phase.</ins></p>
+            <p><ins>In addition, the ~source~ phase indicates that preloading should not take place in the host for this module.</ins></p>
```

Comment:
"Preloading" is not an ECMA262 thing, so this should be a note (ie not normative).

I still question the need for passing phase but this does not impact observable semantics to users, as long as there is note that identity and load is not affected. Let's consider this in a follow up.